test/e2e/aws_util_test.go

```diff
@@ -0,0 +1,312 @@
+					}
+
+					for _, networkInterface := range describeNetworkInterfacesOutput.NetworkInterfaces {
+						if networkInterface.VpcId != nil && *networkInterface.VpcId == vpcID {
+							unassociatedEIPs = append(unassociatedEIPs, struct {
+								PublicIp     string
+								AllocationId string
+							}{
+								PublicIp:     aws.StringValue(address.PublicIp),
+								AllocationId: aws.StringValue(address.AllocationId),
+							})
+						}
+					}
+				} else {
+					// If there's no associated network interface, assume it's an unassociated EIP
+					unassociatedEIPs = append(unassociatedEIPs, struct {
+						PublicIp     string
+						AllocationId string
+					}{
+						PublicIp:     aws.StringValue(address.PublicIp),
+						AllocationId: aws.StringValue(address.AllocationId),
+					})
+				}
+			}
+		} else {
+			// Check if the EIP has the specified tag key and value
+			hasTag := false
+			for _, tag := range address.Tags {
+				if *tag.Key == tagKeyEIP && *tag.Value == tagValueEIP {
+					hasTag = true
+					break
+				}
+			}
+			if hasTag {
+				// Check if the EIP is associated with a network interface in the specific VPC
+				if address.NetworkInterfaceId != nil {
+					describeNetworkInterfacesInput := &ec2.DescribeNetworkInterfacesInput{
+						NetworkInterfaceIds: []*string{address.NetworkInterfaceId},
+					}
+					describeNetworkInterfacesOutput, err := svc.DescribeNetworkInterfaces(describeNetworkInterfacesInput)
+					if err != nil {
+						t.Fatalf("failed to describe network interfaces, %v", err)
+					}
+
+					for _, networkInterface := range describeNetworkInterfacesOutput.NetworkInterfaces {
+						if networkInterface.VpcId != nil && *networkInterface.VpcId == vpcID {
+							associatedEIPs = append(associatedEIPs, struct {
+								PublicIp     string
+								AllocationId string
+							}{
+								PublicIp:     aws.StringValue(address.PublicIp),
+								AllocationId: aws.StringValue(address.AllocationId),
+							})
+						}
+					}
+				} else {
+					associatedEIPs = append(associatedEIPs, struct {
+						PublicIp     string
+						AllocationId string
+					}{
+						PublicIp:     aws.StringValue(address.PublicIp),
+						AllocationId: aws.StringValue(address.AllocationId),
+					})
+				}
+			}
+		}
+	}
+
+	if len(unassociatedEIPs) == 0 {
+		t.Log("No unassociated EIPs found with the specified tag key and value.")
+	} else {
+		t.Log("Unassociated EIPs with the specified tag key and value:")
+		for _, eip := range unassociatedEIPs {
+			t.Logf("Public IP: %s, Allocation ID: %s\n", eip.PublicIp, eip.AllocationId)
+		}
+	}
+
+	// Release each unassociated EIP
+	for _, eip := range unassociatedEIPs {
+		releaseAddressInput := &ec2.ReleaseAddressInput{
+			AllocationId: aws.String(eip.AllocationId),
+		}
+		_, err := svc.ReleaseAddress(releaseAddressInput)
+		if err != nil {
+			t.Errorf("Failed to release EIP %s with Allocation ID: %s\n", eip.PublicIp, eip.AllocationId)
```

Comment:
You don't need `\n`, and you should report the error, that will be important for debugging.

```suggestion
			t.Errorf("Failed to release EIP %s with Allocation ID %s: %v", eip.PublicIp, eip.AllocationId, err)
```

Reply:
Placeholder argument 'eip.PublicIp' has the wrong type '*string' (%s)    so need to add %v

Reply:
./aws_eip_for_nlb_test.go:561:4: (*testing.common).Logf format %s has arg eip.PublicIp of wrong type *string